src/main.cc

```diff
@@ -178,6 +178,25 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_int_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+    (void) loop;
+    (void) w;
+    (void) revents;
+
+    if (pid_file)
+        say_crit("got signal %d - %s", w->signum, strsignal(w->signum));
```

Comment:
I see the same code in the callback above, which is to don't exit silently in the daemonized process (and it is commented). Here I don't understand idea: please, leave a note.

BTW, what we should do when SIGINT is sent to a daemonized process? Or, right, if it is not daemonized, but non-interactive? Please elaborate a proposal.

Reply:
What is strange: I don't see this message in the log after `box.cfg({log = 'tarantool.log'})`.

Reply:
This type of signal is pretty typical for use in apps with interactive mode and it is a bit exotic for interrupting daemons. That is why I meant that in such case we could exit the process with logging in file. As an alternative it is still possible to continue the process execution after interruption and detecting the daemon mode. Meanwhile other daemons like nginx have no message in log if they are interrupted, so they just exit. 

In case Tarantool executes non-interactive, SIGINT affects its output. I see this affection undesirable, so I fix it with next commit.

Reply:
This logic would look good if we would exit after SIGINT in the daemon mode, but according to the code we don't.

Reply:
> Regarding `^C`: it seems, it is controlled by `stty -echoctl` and is out of application's responsibility. Anyway, it is interesting why it does not work for us exactly like for bash.

I guess that bash and Tarantool console just have different SIGINT handlers, that is why the behavior can be different. So if we need to print anything in console after any signal we should add this behavior to handler manually.

Reply:
I have changed this behavior for daemon mode, now it is exiting after recieving SIGINT with `0` code.

Reply:
But why we exit without any message into logs? The comment from `signal_cb()` does not convinced you that it is important?

My initial comment was about lack of a comment here.

Reply:
> What is strange: I don't see this message in the log after `box.cfg({log = 'tarantool.log'})`.

How about this? Did you check it?

Reply:
You can see this message in log if you have specified the pid file. There is a parameter in box.cfg which calls `pid_file` and it has nil by default.

Reply:
I see now, fine.